web/src/settings_components.ts

```diff
@@ -1348,3 +1391,93 @@ export function initialize_disable_btn_hint_popover(
     }
     tippy.default(util.the($btn_wrapper), tippy_opts);
 }
+
+const group_setting_widget_map = new Map<string, GroupSettingPillContainer | null>([
+    ["can_manage_group", null],
+    ["new_group_can_manage_group", null],
+]);
+
+export function get_group_setting_widget(setting_name: string): GroupSettingPillContainer | null {
+    const pill_widget = group_setting_widget_map.get(setting_name);
+
+    if (pill_widget === undefined) {
+        blueslip.error("No dropdown list widget for property", {setting_name});
+        return null;
+    }
+
+    return pill_widget;
+}
+
+export function set_group_setting_widget_value(
+    property_name: string,
+    property_value: GroupSettingType,
+): void {
+    const pill_widget = get_group_setting_widget(property_name);
+    assert(pill_widget !== null);
+    pill_widget.clear();
+
+    if (typeof property_value === "number") {
+        const user_group = user_groups.get_user_group_from_id(property_value);
+        if (user_group.name === "role:nobody") {
+            return;
+        }
+        user_group_pill.append_user_group(user_group, pill_widget);
+    } else {
+        for (const setting_sub_group_id of property_value.direct_subgroups) {
+            const user_group = user_groups.get_user_group_from_id(setting_sub_group_id);
+            if (user_group.name === "role:nobody") {
+                continue;
+            }
+            user_group_pill.append_user_group(user_group, pill_widget);
+        }
+        for (const setting_user_id of property_value.direct_members) {
+            const user = people.get_user_by_id_assert_valid(setting_user_id);
+            user_pill.append_user(user, pill_widget);
+        }
+    }
+}
+
+export type GroupSettingType = z.output<typeof group_setting_type_schema>;
+
+type group_setting_name = "can_manage_group";
+
+export function create_group_setting_widget({
+    $pill_container,
+    setting_name,
+    group,
+}: {
+    $pill_container: JQuery;
+    setting_name: group_setting_name;
+    group?: UserGroup;
+}): void {
+    const pill_widget = group_setting_pill.create_pills($pill_container, setting_name);
+    const opts = {
+        setting_name,
+        group,
+    };
+    group_setting_pill.set_up_pill_typeahead({pill_widget, $pill_container, opts});
+
+    if (group === undefined) {
+        group_setting_widget_map.set("new_group_" + setting_name, pill_widget);
+    } else {
+        group_setting_widget_map.set(setting_name, pill_widget);
+    }
```

Comment:
I'm not sure this `new_group` pattern is a great one; why does this need to be a different element?

Seems OK to consider cleaning up in a follow-up PR regardless.

Reply:
We need to store the pill-widget for new group and existing group separately, because the widget for new group is set-up once and is not set-up again while navigating between group edit and group creation UI.

Reply:
But can see if we can have a better approach as a follow-up.

Reply:
Yeah, what I'm thinking is maybe we wouldn't need this if we just set it up whenever entering the panel...